### ISSUE TITLE
Napari >0.4.5 window non-responsive if created on non-primary display.

### ISSUE THREAD

**Issue by Remote-Radical** [napari/napari#2732]
## 🐛 Bug

Starting from version 0.4.6 launching napari will result in white, unresponsive window, if created on secondary display. 
This is certainly tied to the the ability to save window position, which was introduced in 0.4.6.
It could also be linked to the case of the secondary screen being in portrait orientation.
EDIT: Issue is identical, regardless of screen orientation

## To Reproduce

1. Launch napari on primary screen
2. Move napari window to second (non-primary) display
3. Close napari window.
4. Launch napari
5. Napari launches a white, unresponsive window in the previous position on the secondary screen. Window needs to be terminated.

Steps to fix the behavior:

1. Close napari
2. Disable all non-primary displays
3. Launch napari
4. napari launches correctly on the primary screen.

## Expected behavior

Napari should work as expected on systems with multiple displays.

## Environment

napari: 0.4.7
Platform: Windows-10-10.0.19041-SP0
Python: 3.8.10 | packaged by conda-forge | (default, May 11 2021, 06:25:23) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.20.3
SciPy: 1.6.3
Dask: 2021.05.0
VisPy: 0.6.6

OpenGL:
  -GL version:  4.6.0 NVIDIA 462.31
  -MAX_TEXTURE_SIZE: 32768

Screens:
  -screen 1: resolution 2560x1440, scale 1.0
  -screen 2: resolution 1440x2560, scale 1.0

Plugins:
  -console: 0.0.3
  -svg: 0.1.5

## Warning logged

WARNING: QWindowsWindow::setGeometry: Unable to set geometry 1440x2537+2560-617 (frame: 1456x2576+2552-648) on QWidgetWindow/"_QtMainWindowClassWindow" on "\\.\DISPLAY1". Resulting geometry: 1440x1421+2560-617 (frame: 1456x1460+2552-648) margins: 8, 31, 8, 8 minimum size: 488x519 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=504,558 maxtrack=0,0)

**Comment by goanpeca** [comment]
Hi @Remote-Radical, thanks for the report. 

>it could also be linked to the case of the secondary screen being in portrait orientation.

Could you double check this option?

While I get my hands on an external monitor I will check using ipad/iphone and see if I can reproduce it.

**Comment by Remote-Radical** [comment]
> > it could also be linked to the case of the secondary screen being in portrait orientation.
> 
> Could you double check this option?

I have verified that the issue is identical regardless of display alignment and orientation.

**Comment by goanpeca** [comment]
>I have verified that the issue is identical regardless of display alignment and orientation.

Thanks :-)

**Comment by goanpeca** [comment]
@Remote-Radical what version of Qt are you using?

I have not been able to reproduce this on OSX with the latest napari. @napari/core-devs could anyone please check if you can reproduce this, or if it a Windows specific Issue?

**Comment by alisterburt** [comment]
also unable to reproduce on master/4.10, environment info below

```sh

napari: 0.4.11.dev22+g955da0af
Platform: macOS-11.2.1-x86_64-i386-64bit
System: MacOS 11.2.1
Python: 3.9.5 | packaged by conda-forge | (default, Jun 19 2021, 00:27:35)  [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.6.6

OpenGL:
  - GL version:  2.1 Metal - 71.0.7
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1440x900, scale 2.0

Plugins:
  - console: 0.0.3
  - scikit-image: 0.4.11.dev22+g955da0af
  - svg: 0.1.5
```

**Comment by nclack** [comment]
I don't see the unresponsive window but I do see some warnings.

```
WARNING: QWindowsWindow::setGeometry: Unable to set geometry 2176x1366-2088+831 (frame: 2198x1422-2099+786) on QWidgetWindow/"_QtMainWindowClassWindow" on "\\.\DISPLAY2". Resulting geometry: 2611x1640-2086+838 (frame: 2633x1696-2097+793) margins: 11, 45, 11, 11 minimum size: 488x539 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=998,1134 maxtrack=0,0)
17:03:05
WARNING QWindowsWindow::setGeometry: Unable to set geometry 2176x1366-2088+831 (frame: 2198x1422-2099+786) on QWidgetWindow/"_QtMainWindowClassWindow" on "\\.\DISPLAY2". Resulting geometry: 2611x1640-2086+838 (frame: 2633x1696-2097+793) margins: 11, 45, 11, 11 minimum size: 488x539 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=998,1134 maxtrack=0,0)
```

I see the same behavior in the current master and in 0.4.7.  My info:

```
napari: 0.4.7
Platform: Windows-10-10.0.19043-SP0
Python: 3.8.10 (default, May 19 2021, 13:12:57) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.7.0

OpenGL:
  - GL version:  4.6.0 NVIDIA 466.77
  - MAX_TEXTURE_SIZE: 32768

Screens:
  - screen 1: resolution 3840x2160, scale 1.0
  - screen 2: resolution 1800x3200, scale 1.0

Plugins:
  - console: 0.0.3
  - svg: 0.1.5
```

**Comment by nclack** [comment]
I should have mentioned, the warnings only show up when starting napari on the second display.

**Comment by jni** [comment]
OMG 👋 👋 👋 @nclack!!! 😊 😊 😊 

@Remote-Radical would you mind attempting to reproduce your issue on a more recent napari and/or another machine? It seems hard to reproduce and thus to fix...

Having said that @nclack given your setup maybe you could investigate the warnings and how one might be able to get rid of them? Fixing the warnings might fix the issue in other users' cases.

**Comment by jxchen01** [comment]
I encountered similar issue. I have a new laptop, the first time I install and use napari on this laptop is when the laptop connected to a secondary display at my office. After getting back home, where I don't have the secondary display, I cannot start napari anymore on the laptop, specifically the window does not show up and got the following message:

```
WARNING: QWindowsWindow::setGeometry: Unable to set geometry 1924x1214-6218+212 (frame: 1946x1270-6229+167) on QWidgetWindow/"_QtMainWindowClassWindow" on "\\.\DISPLAY1". Resulting geometry: 1924x1098-6218+212 (frame: 1946x1154-6229+167) margins: 11, 45, 11, 11 minimum size: 608x549 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=1238,1154 maxtrack=0,0)
20:16:11 WARNING QWindowsWindow::setGeometry: Unable to set geometry 1924x1214-6218+212 (frame: 1946x1270-6229+167) on QWidgetWindow/"_QtMainWindowClassWindow" on "\\.\DISPLAY1". Resulting geometry: 1924x1098-6218+212 (frame: 1946x1154-6229+167) margins: 11, 45, 11, 11 minimum size: 608x549 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=1238,1154 maxtrack=0,0)
```

Tried to re-install napari on a different conda environment, still not work.

napari verion = 0.4.12
Windows 10

the same issue for any version after  0.4.5

**Comment by goanpeca** [comment]
@jxchen01 what OS are you working on?

**Comment by jni** [comment]
@jxchen01 please note that until we fix this you can run `napari --reset` on the command line to clear the settings, including the screen position settings that are causing the problem.

**Comment by psobolewskiPhD** [comment]
@melonora You were playing with external monitors and napari windows, did you run into this? Is this bug still a thing?

**Comment by melonora** [comment]
From what I can tell this bug does not exist anymore. It was related to the window positions being outside the screen geometry in case of the secondary screen being disconnected. This was fixed with #5915. I will close this one. 
There are some other issues, but for those I already opened other issues. I will therefore close this issue.
